src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs

```diff
@@ -256,7 +265,7 @@ public WebSocketsTransport(ILoggerFactory loggerFactory)
 
                     // We intentionally don't pass _transportCts.Token to CloseOutputAsync. The token can be cancelled
                     // for reasons not related to webSocket in which case we would not close the websocket gracefully.
-                    await _webSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, null, CancellationToken.None);
+                    await _webSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, statusDescription: null, cancellationToken: cancellationToken);
```

Comment:
I think passing the token to CloseOutputAsync is incorrect. if you do that you might leave the connection is undefined state - the transport will be possibly still connected (because closing was cancelled) but the connection will transition to the `Disconnected` state and there will be no way to ever close the websocket (if you called DisposeAsync again it will no op because it is in the `Disconnected` state). Also, see the comment above.